### ISSUE TITLE
Using "--use-netrc" option fails to download archive from private repository release

### ISSUE THREAD

**Issue by mpdifran** [Carthage/Carthage#3043]
* carthage install method: [ ] .pkg, [x] homebrew, [ ] source
* `which carthage`: `/usr/local/bin/carthage`
* `carthage version`: 0.35.0
* `xcodebuild -version`: Xcode 11.7 Build version 11E801a
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? No
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? No
* Are you using `--new-resolver`? No
* Are you using `--use-netrc`? Yes

**Cartfile**
```
binary "FaireProtobuf.json" ~> 3.15
```

**FaireProtobuf.json**
```
{
    "3.15": "https://github.com/Faire/ios-protobuf/releases/download/3.15/FaireProtobuf.framework.zip"
}
```

**Carthage Command**
```
carthage update --platform iOS --use-netrc
```

**Carthage Output**
```
*** Downloading binary-only framework FaireProtobuf at "file:///Users/markdifranco/Sandbox/ios-FaireCore/FaireProtobuf.json"
*** xcodebuild output can be found in /var/folders/xc/z9pwbp6123nf7br09dzt0h8w0000gn/T/carthage-xcodebuild.oVJCfO.log
*** Downloading binary-only framework FaireProtobuf at "file:///Users/markdifranco/Sandbox/ios-FaireCore/FaireProtobuf.json"
*** Downloading FaireProtobuf.framework binary at "3.15.0"
A shell task (/usr/bin/env unzip -uo -qq -d /var/folders/xc/z9pwbp6123nf7br09dzt0h8w0000gn/T/carthage-archive.j8kpT1 /Users/markdifranco/Library/Caches/org.carthage.CarthageKit/binaries/FaireProtobuf/3.15.0/FaireProtobuf.framework.zip) failed with exit code 9:
[/Users/markdifranco/Library/Caches/org.carthage.CarthageKit/binaries/FaireProtobuf/3.15.0/FaireProtobuf.framework.zip]
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /Users/markdifranco/Library/Caches/org.carthage.CarthageKit/binaries/FaireProtobuf/3.15.0/FaireProtobuf.framework.zip or
        /Users/markdifranco/Library/Caches/org.carthage.CarthageKit/binaries/FaireProtobuf/3.15.0/FaireProtobuf.framework.zip.zip, and cannot find /Users/markdifranco/Library/Caches/org.carthage.CarthageKit/binaries/FaireProtobuf/3.15.0/FaireProtobuf.framework.zip.ZIP, period.
```

**Actual outcome**
Carthage failed to download the framework archive. Inspecting the download, it's simply a text file containing `Not Found`.

**Expected outcome**
Download the framework archive.

**Notes**
The `.netrc` file is set up to use an account with 2FA enabled. I've generated a Personal Access Token with the private repo scope, and put that in the `.netrc`. I've verified the `.netrc` file works using curl.

**Comment by mpdifran** [comment]
@mollyIV I noticed you implemented #2774 , any insights here? Am I referencing my binary file in the JSON correctly?

**Comment by alexfu** [comment]
FWIW, the netrc implementation [only does Basic auth](https://github.com/Carthage/Carthage/blob/84d068b/Source/CarthageKit/Netrc.swift#L31).

Downloading Github release assets from a private repo requires an auth token, i.e. `Authorization: token <TOKEN>` .
